src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1628,6 +1634,24 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
     return false;
 }
 
+bool OutputStateMachineEngine::_ParseDirectory(const std::wstring_view string,
+                                               std::wstring& uri) const
+{
+    if (string.length() <= 8)
+    {
+        return false;
+    }
+
+    const auto prefix = string.substr(0, 7);
+    if (!prefix.compare(L"file://") == 0)
+    {
+        return false;
+    }
+
+    uri = std::wstring(string.substr(7, string.length() - 7));

```

Comment:
Is the plan to later add hostname validation, so we can filter-out SSH-sourced, Docker-sourced, and similar remote paths? That would eliminate a source of false-positives for the 'Is Valid Path' test, assuming existing code that generates OSC 7 is including the hostname. Hostname is described as "should" in the [Terminal.app specification](https://github.com/microsoft/terminal/issues/3158#issuecomment-578550385).

Reply:
As much as I want to do this, there’s certain difficulty. First for my WSL distro, there’s no HOSTNAME in the env. There’s only HOST which is the computer name of the Windows OS. Similar for PowerShell. The concept of HOSTNAME does not to fit in the world of Windows. 

In the *nix world the path is interchangeable between the local machine and remote machine. However if you’re sshing to a Linux machine on Windows, the paths on the remote machine are guaranteed to be invalid on the local machine. My bash/zsh scripts actually make great use of the nature. The two types of the URIs it can generate are “c:\path” and “\\wsl$\path”. None of them is a valid *nix path. Pretty ironic now I put it this way. But it sure works. In the Windows world I can not find a good place for HOSTNAME.

Reply:
Weird. I just checked my WSL2 Ubuntu, and I have `$HOSTNAME` but not `$HOST`. Both Git Bash and MSYS2 also have `$HOSTNAME`. In the Windows environment, it's `$COMPUTERNAME` (`$env:COMPUTERNAME` in PowerShell), and that env-var is visible in Git Bash and MSYS2 as well.

Either way, it's up to the person enabing OSC 7 to ensure they enable it to generate a hostname.

It would also be reasonable to accept hostname "" as "trust me, it's localhost" for simplicity. The main need here is to filter _out_ paths coming from remote machines, whether it's bash over SSH or PowerShell remoting, without relying on the remote host generating paths that are invalid locally.

Reply:
Personally I’d like to stick to the “do not probe the path passed in” policy because it gives us more flexibility without bringing too much complexity to the terminal itself. Of course I’m no expert at VT, or terminal, or anything really. If there’s a better way most people can agree on, I’m totally cool with changing it.

Reply:
I've just realised this is a bug: It's trimming off 7 characters, `file://`, but in fact the path starts after `file://<hostname>/`. So this as-is won't work with OSC 7 commands that work according to the Terminal.app help text, nor the [formerly-proposed spec at terminal-wg](https://gitlab.freedesktop.org/terminal-wg/specifications/-/blob/5f6bd72258b3d62e9cbc01908c32783d025f2893/proposals/working-path-notifications.md).

I don't consider it 'probing' the path to be correctly separating the part before the path, and since that will give you the shell-provided hostname anyway, ignoring it if it isn't "" or `$COMPUTERNAME` seems reasonable to me (and is what's described in the Terminal.app help text and the proposed spec)

Reply:
Yes I intended to ignore anything about HOSTNAME, because as I said in
other comments, I can not find a meaningful thing to do with it. I wasn’t really trying to follow the spec when I wrote this. But you’re right. Following the spec helps the community in some way. 

Separating the HOSTNAME of course does not count as probing. But “filtering out SSH-sourced, Docker-sourced, and similar remote paths” based on that is something I chose to avoid.

Reply:
That's why the spec was "Use hostname to recognise remote paths". It doesn't say you need to filter-out using that knowledge; and now that I know that `file://wsl$/Ubuntu` is the correct URL for the WSL mount, I don't think we can actually _ignore_ such paths here after all, and will have to rely on the 'is valid path' check when using it value.

Reply:
Sadly `startingDirectory` does not accept `file://`. We just have to live with it.